### ISSUE TITLE
[Go Program Gen] Examples quality pass

### ISSUE THREAD

**Issue by EvanBoyle** [pulumi/pulumi#4783]
Once we have all of the test cases passing in pulumi/pulumi, let's spot check the examples in a few providers for obvious bugs. We've already seen a few cases of bad types, incomplete recursive traversal for input intrinsic, etc.

Deferring until the test cases are all passing will allow some of these issues to shake out naturally.

**Comment by EvanBoyle** [comment]
Note, these are the actionable ones. There are many that are the result of TF errors, type errors, and in the general category of bad docs that are a problem for other languages as well.

**Comment by EvanBoyle** [comment]
Here's the list of all diagnostic errors with TF source (all languages) for posterity's sake and in case we want to do some further analysis: 

https://gist.github.com/EvanBoyle/072f4794472c624835870f99cee59528

**Comment by EvanBoyle** [comment]
I combed through the first 10% of examples thoroughly, and then spot checked another 15%. Concrete issues in a gist below. 

TLDR is that issues are overall very minor and mostly easy to fix. I'll create individual issues based on the results. 

https://gist.github.com/EvanBoyle/3344dceb3f30b1c4022d5dfb84bb40df

**Comment by EvanBoyle** [comment]
Here is the list of errors in existing examples that we're generating. Including warnings/formatting bugs for completeness:

- (warning) redundant array type elm decls: https://github.com/pulumi/pulumi/issues/4897
- (formatting) Multi-line string formatting: https://github.com/pulumi/pulumi/issues/4898
- Wrap invoke results in pulumi types within `ctx.Export` https://github.com/pulumi/pulumi/issues/4899
- Correct usage of `:=` for assignment to `err` https://github.com/pulumi/pulumi/issues/4900
- Non-optional type for invoke requiring pointer primitive https://github.com/pulumi/pulumi/issues/4901
- Fix pulumi.Map types https://github.com/pulumi/pulumi/issues/4903

The only one of these that I'm concerned about fixing immediately is the pulumi.Map types. The others are all fairly minor and pretty obvious to fix for consumers. Open to feedback here though.

**Comment by EvanBoyle** [comment]
As for closing the coverage gap (currently at 91% of examples for AWS) there's one larger issue for the NYIs:

https://github.com/pulumi/pulumi/issues/4905

And one remaining issue for some invalid variable names that cause gofmt to fail:

https://github.com/pulumi/pulumi/issues/4906

**Comment by EvanBoyle** [comment]
After discussing with @leezen we decided to fix https://github.com/pulumi/pulumi/issues/4903 (MapTypes) and leave the remaining issues for the time being. I'll be switching over to the runtime workstream. Either I or another member of the team can pick up the remaining work outlined in these issues at a later date.
